File: bankingonrails/app/models/kycprofile.rb
class KycProfile < ApplicationRecord
  enum Status: [:Pending, :Verified, :Rejected, :Expired]


  has_many :Customer, class_name: 'Customer'
  has_many :IdentityDocuments, class_name: 'IdentityDocument'
  has_many :RiskAssessments, class_name: 'RiskAssessment'
  has_many :Screenings, class_name: 'ScreeningResult'

end

File: bankingonaspdotnet/Domain/KycProfile.cs
using bankingonaspdotnet.Contracts;

namespace bankingonaspdotnet.Domain;

public class KycProfile
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long? KycprofileId { get; set; } 
 public virtual string? ProfileId { get; set; } 
 public virtual DateOnly? LastReviewedOn { get; set; } 
public virtual Customer? Customer { get; set; } 
public virtual ICollection<IdentityDocument> IdentityDocuments { get; set; } = new List<IdentityDocument>();
public virtual ICollection<RiskAssessment> RiskAssessments { get; set; } = new List<RiskAssessment>();
public virtual ICollection<ScreeningResult> Screenings { get; set; } = new List<ScreeningResult>();
 public virtual KycStatus? Status { get; set; } 

    public static KycProfile FromRequest(KycProfileRequest request) {
        return new KycProfile {
            Id = request.Id,
            ProfileId = request.ProfileId,
            LastReviewedOn = request.LastReviewedOn,
            Status = request.Status,
        };
    }
}
